golos.referral/src/golos.referral.cpp

```diff
@@ -0,0 +1,24 @@
+#include "golos.referral/golos.referral.hpp"
+
+namespace golos {
+
+using namespace eosio;
+
+void referral::crtreferral(name referrer, name referral, uint32_t percent,
+                           uint64_t expire, uint64_t breakout) {
+    require_auth(referrer);
```

Comment:
Verifying permission only from the referrer gives him the opportunity to add as many referrals as he wants and thus receive their income without their consent.
@afalaleev: it is necessary to decide who and how can connect users to the referral program.

Reply:
Moved to https://github.com/GolosChain/golos-smart/issues/374